pkg/storage/http/checker.go

```diff
@@ -0,0 +1,34 @@
+package http
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+)
+
+// Exists checks for a specific version of a module
+func (s *ModuleStore) Exists(ctx context.Context, module, vsn string) (bool, error) {
+	var op errors.Op = "http.Exists"
+	ctx, span := observ.StartSpan(ctx, op.String())
+	defer span.End()
+
+	req, _ := http.NewRequest(http.MethodHead, s.moduleRoot(module)+vsn+".mod", nil)
```

Comment:
would you mind making a function that does all the appending for you? so you could write something like `s.modulePath(module, version, "mod")`